src/Grpc.AspNetCore.Server/InterceptorCollection.cs

```diff
@@ -28,21 +28,44 @@ namespace Grpc.AspNetCore.Server
     /// </summary>
     public class InterceptorCollection : Collection<InterceptorRegistration>
     {
+        internal InterceptorCollection()
+        {
+        }
+
         /// <summary>
         /// Add an interceptor to the end of the pipeline.
         /// </summary>
         /// <typeparam name="TInterceptor">The interceptor type.</typeparam>
         /// <param name="args">The list of arguments to pass to the interceptor constructor when creating an instance.</param>
         public void Add<TInterceptor>(params object[] args) where TInterceptor : Interceptor
         {
-            Add(new InterceptorRegistration(typeof(TInterceptor), args));
+            Add(typeof(TInterceptor), args);
+        }
+
+        /// <summary>
+        /// Add an interceptor to the end of the pipeline.
+        /// </summary>
+        /// <param name="interceptorType">The interceptor type.</param>
+        /// <param name="args">The list of arguments to pass to the interceptor constructor when creating an instance.</param>
+        public void Add(Type interceptorType, params object[] args)
+        {
+            if (interceptorType == null)
+            {
+                throw new ArgumentNullException(nameof(interceptorType));
+            }
+            if (!interceptorType.IsSubclassOf(typeof(Interceptor)))
+            {
+                throw new ArgumentException($"Type must inherit from {typeof(Interceptor).FullName}.", nameof(interceptorType));
+            }
+
+            Add(new InterceptorRegistration(interceptorType, args));
         }
 
         /// <summary>
         /// Append a set of interceptor registrations to the end of the pipeline.
         /// </summary>
         /// <param name="registrations">The set of interceptor registrations to add.</param>
-        public void AddRange(IEnumerable<InterceptorRegistration> registrations)
+        internal void AddRange(IEnumerable<InterceptorRegistration> registrations)
```

Comment:
InterceptorRegistration can only be create by Add. As far as I can tell this is never useful.

Reply:
The original idea for adding this originally was that you can create a collection of interceptors that can be added to different services.